actors/miner/src/lib.rs

```diff
@@ -1924,8 +1924,7 @@ impl Actor {
 
         if !params.aggregate_proof.is_empty() {
             // Aggregate fee is paid on the sectors successfully proven,
-            // but without regard to data activation which could subsequently fail
-            // and prevent sector activation.
+            // but without regard to data activation success.
```

Comment:
```suggestion
            // but without regard to data activation which may have subsequently failed
            // and prevented sector activation.
```

I can't speak for @NemanjaLu92 but I'm interpreting his comments about this block in Slack as being confused about the comment suggesting some future event while activation seems to have already occurred. My initial read of the comment had that interpretation at least, I think because the language can (but not necessarily) be interpreted as discussing future events.

"may have", "failed" and "prevented" makes the tense clear.

Reply:
Data activation hasn't happened yet, it's immediately below this at 1946. The fee is charged before data activation to make it clear that it doesn't depend on the results.

Reply:
Maybe we have a language problem then? This whole section looks like "data activation": https://github.com/filecoin-project/builtin-actors/blob/ea7c45478751bd0fe12d0d374abc8fdc9341bfea/actors/miner/src/lib.rs#L1895-L1905 which is followed by "activation" of sector info. 1946 is "notification". When it's viewed from the market actor, does it become another kind of "data activation"? Would "deal activation" be alternative language for this?

Reply:
Ah yes, sorry. Around 1897 is data activation (in the FIL+ sense), which _is_ required to succeed. I'd rather not use the term "deal" (which is on its way out of the miner actor), but "notification" would be fine.

The terminology confusion here arises from refactors to support DDO while also supporting the old methods. Data activation is the generic term adopted for either or both claiming Fil+ allocations and activating deals in the legacy flows where both must succeed. In this v3 flow where the market notification need not succeed, it's not really part of data activation anymore.

Reply:
```suggestion
            // but without regard to notification success.
```

so that's enough then?

Reply:
I'm so sorry, I have been dumb and confused us all.

Let's look at the code immediately below. The input to the fee calculation is the number of proven sectors (i.e. number of proofs aggregate), not the activated sectors. The edit proposed originally in this PR is good, but the suggested change that you made at the top of this thread is even better. Let's go with that.

(Notifications are so irrelevant here that they're not even worth mentioning.)